refactor(platform-browser): replace `isPlatformServer` with `ngServerMode`

arturovt opened the pull request:
In this commit, we switch from using the `isPlatformServer` runtime call to the `ngServerMode`.

Note: constructors haven't been touched in order to prevent any breaking changes for the public API.

thePunderWoman commented:
@arturovt Can you resolve the conflicts here and take a look at the failing tests? I suspect they are related to this change.

JeanMeche commented:
`isPlatformServer` is public while `ngServerMode` isn't.

thePunderWoman reviewed (dismissed):
LGTM

I think we can move forward with this. @arturovt Can you do the cleanup on this?

arturovt commented:
@thePunderWoman yes, give me some time (w/o closing this PR), I'm just doing cleanups on other PRs as well...

arturovt committed:
refactor(platform-browser): replace `isPlatformServer` with `ngServerMode`

In this commit, we switch from using the `isPlatformServer` runtime call to the `ngServerMode`.

Note: constructors haven't been touched in order to prevent any breaking changes for the public API.

arturovt commented:
@thePunderWoman the `core/test:test` is green locally for me, could you approve the CI run?

JeanMeche commented on packages/platform-browser/src/dom/dom_renderer.ts:
```diff
@@ -165,7 +165,11 @@ export class DomRendererFactory2 implements RendererFactory2, OnDestroy {
 
-    if (this.platformIsServer && type.encapsulation === ViewEncapsulation.ShadowDom) {
+    if (
+      typeof ngServerMode !== 'undefined' &&
```
I don't think the optimizer would be able to infer the value of `this.platformIsServer` as `false` and remove the condition.

arturovt replied:
Exactly.

thePunderWoman reviewed (approved):
LGTM

thePunderWoman commented:
This PR was merged into the repository by commit 9a9511ace8212f13d637be24990036c93fac48e2.

The changes were merged into the following branches: main, 20.0.x
